packages/fast-colors/src/luminosity.ts

```diff
@@ -0,0 +1,53 @@
+import Chroma from "chroma-js";
+
+/**
+ * Type definition for a luminosity switch
+ */
+export type LuminositySwitch = (a: any, b: any) => any;
+
+/**
+ * Adjust a color to a specific luminosity. This function is almost a direct copy of
+ * https://github.com/gka/chroma.js/blob/master/src/io/luminance.coffee, except that
+ * it accepts a rounding function. This is necessary to prevent contrast ratios being slightly below
+ * their target due to rounding RGB channel values the wrong direction.
+ */
+export function luminance(targetLuminance: number, sourceColor: Chroma, round?: (value: number) => number): number[] {
+    const sourceLuminosity: number = sourceColor.luminance();
+    const fidelity: number = 1e-7;
+    let maxItterations: number = 20;
+
+    function test(low: Chroma, high: Chroma): any {
```

Comment:
do you still want to call this function test? looks like its returning a middle value so could we do `getMiddleLuminosity` or some such thing?